hack/verify-staticcheck.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+kube::golang::verify_go_version
+
+FOCUS="${1:-}"
+
+# See https://staticcheck.io/docs/checks
+CHECKS=(
+  "all"
+  "-S1*"   # Omit code simplifications for now.
+  "-ST1*"  # Mostly stylistic, redundant w/ golint
+)
+export IFS=','; checks="${CHECKS[*]}"; unset IFS
+
+# Packages to ignore due to bugs in staticcheck
+# NOTE: To ignore issues detected a package, add it to the .staticcheck_failures blacklist
+IGNORE=(
+  "vendor/k8s.io/kubectl/pkg/cmd/edit/testdata"  # golang/go#24854, dominikh/go-tools#565
+)
+export IFS='|'; ignore_pattern="^(${IGNORE[*]})\$"; unset IFS
+
+# Ensure that we find the binaries we build before anything else.
+export GOBIN="${KUBE_OUTPUT_BINPATH}"
+PATH="${GOBIN}:${PATH}"
+
+# Install staticcheck from vendor
+echo 'installing staticcheck from vendor'
+go install k8s.io/kubernetes/vendor/honnef.co/go/tools/cmd/staticcheck
+
+cd "${KUBE_ROOT}"
+
+# Check that the file is in alphabetical order
+failure_file="${KUBE_ROOT}/hack/.staticcheck_failures"
+kube::util::check-file-in-alphabetical-order "${failure_file}"
+
+export IFS=$'\n'
+# NOTE: when "go list -e ./..." is run within GOPATH, it turns the k8s.io/kubernetes
+# as the prefix, however if we run it outside it returns the full path of the file
+# with a leading underscore. We'll need to support both scenarios for all_packages.
+all_packages=()
+while IFS='' read -r line; do
+  all_packages+=("./$line")
+done < <( hack/make-rules/helpers/cache_go_dirs.sh .make/all_go_dirs.mk |
```

Comment:
In this case you should write to your own temp file, not .make/all_go_dirs.mk (that's mine :)

Reply:
Went with `_tmp/all_go_dirs` so we can share it with the other verify scripts. I'll do a follow up PR to move them to this function.